content/en/docs/concepts/cluster-administration/node-autoscaling.md

```diff
@@ -0,0 +1,306 @@
+resizing the underlying cloud provider resource group.
+
+Additional context:
+
+* Documentation overview:
+  https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/README.md
+* Supported cloud providers:
+  https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/README.md#faqdocumentation
+* Cluster Autoscaler FAQ:
+  https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md
+* Contact: https://github.com/kubernetes/community/tree/master/sig-autoscaling#contact
+
+#### Karpenter
+
+Karpenter auto-provisions Nodes based on [NodePool](https://karpenter.sh/docs/concepts/nodepools/)
+configuration provided by the cluster operator. Karpenter handles all aspects of node lifecycle, not
+just autoscaling. It works directly with individual cloud provider resources (most commonly
+individual Virtual Machines), and doesn't rely on cloud provider resource groups.
+
+Additional context:
+
+* Documentation: https://karpenter.sh/
+* Supported cloud providers:
+  https://github.com/kubernetes-sigs/karpenter?tab=readme-ov-file#supported-cloudproviders
+* Karpenter FAQ: https://karpenter.sh/docs/faq/
+* Contact:
+  https://github.com/kubernetes-sigs/karpenter#community-discussion-contribution-and-support
+
+#### Implementation comparison
+
+Main differences between Cluster Autoscaler and Karpenter:
+
+* Cluster Autoscaler provides features related to just Node autoscaling. Karpenter has a wider
+  scope, and also provides features intended for managing Node lifecycle altogether (for example
+  grouping Nodes into pools, handling Node upgrades via disruption).
+* Cluster Autoscaler doesn't support auto-provisioning, the Node groups it can provision from have
+  to be pre-configured. Karpenter supports auto-provisioning, so the user only has to configure a
+  set of constraints for the provisioned Nodes instead of fully configuring homogenous groups.
+* Cluster Autoscaler provides integrations with numerous cloud providers, including smaller and less
+  popular providers. Karpenter currently only provides integrations with AWS and Azure.
+
+## Configure workloads for autoscaling {#workload-config}
+
+Normally, workloads don't have to be additionally configured to work with Node autoscaling, beyond
+just configuring the autoscaler. However, autoscalers provide the ability to change their behavior
+based on workload characteristics. The configuration options documented here are supported by all
+official autoscalers. Thanks to this, the workloads can be portable between clusters using different
+autoscalers. Additionally, the autoscalers provide their own, non-portable configuration options
+(see autoscaler-specific documentation).
+
+### Configure a workload not to be disrupted by consolidation
+
+{{< feature-state state="stable" for_k8s_version="v1.31" >}}
+
+You can use annotation `autoscaling.kubernetes.io/do-not-disrupt: true` on a Pod to prevent it being
```

Comment:
Agree with @sftim, it might be worth breaking out the stuff in here that has to do with API alignment so that we don't have to hold this docs change based on the proposed API changes. We obviously still need to discuss this annotation and also get this registered elsewhere in upstream